sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -235,6 +235,69 @@ case class CreateMap(children: Seq[Expression]) extends Expression {
   override def prettyName: String = "map"
 }
 
+/**
+ * Returns a catalyst Map containing the two arrays in children expressions as keys and values.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(keys, values) - Creates a map with a pair of the given key/value arrays. All elements
+      in keys should not be null""",
+  examples = """
+    Examples:
+      > SELECT _FUNC_([1.0, 3.0], ['2', '4']);
+       {1.0:"2",3.0:"4"}
+  """, since = "2.4.0")
+case class CreateMapFromArray(left: Expression, right: Expression)
+    extends BinaryExpression with ExpectsInputTypes {
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(ArrayType, ArrayType)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    (left.dataType, right.dataType) match {
+      case (ArrayType(_, cn), ArrayType(_, _)) =>
+        if (!cn) {
+          TypeCheckResult.TypeCheckSuccess
+        } else {
+          TypeCheckResult.TypeCheckFailure("All of the given keys should be non-null")
+        }
+      case _ =>
+        TypeCheckResult.TypeCheckFailure("The given two arguments should be an array")
+    }
+  }
+
+  override def dataType: DataType = {
+    MapType(
+      keyType = left.dataType.asInstanceOf[ArrayType].elementType,
+      valueType = right.dataType.asInstanceOf[ArrayType].elementType,
+      valueContainsNull = left.dataType.asInstanceOf[ArrayType].containsNull)
+  }
+
+  override def nullable: Boolean = false
+
+  override def nullSafeEval(keyArray: Any, valueArray: Any): Any = {
+    val keyArrayData = keyArray.asInstanceOf[ArrayData]
```

Comment:
Shouldn't you detect duplicities first?

Reply:
Could you please let us know where this specification is described or is derived from? It is not written [here](https://prestodb.io/docs/current/functions/map.html).

Reply:
Although it's not specified, duplicated key can lead to non-determinism of returned values in future. Currently, `GetMapValueUtil.getValueEval` returns a value for the first key in the map, but there is TODO to change O(n) algorithm. So I'm wondering how it would behave if some hashing was introduced.

Reply:
I see. According to current Spark implementation, for example, `CreateMap` allows us to have duplicated key.
It would be good to discuss such a behavior change in another PR. WDYT?

Reply:
Yeah, we don't have to change it now. But I would like to agree on a consistent approach for the new functions, since this is also related to [SPARK-23934](https://issues.apache.org/jira/browse/SPARK-23934) and [SPARK-23936](https://issues.apache.org/jira/browse/SPARK-23936).

Reply:
I would like to err on the safe side here. `CreateMap` should be fixed IMO.